java/c/src/test/java/org/apache/arrow/c/DictionaryTest.java

```diff
@@ -293,4 +315,38 @@ public void testVectorLoadUnloadOnStructVector() {
       }
     }
   }
+
+  @Test
+  public void testVectorLoadUnloadOnStructVectorWithInline() {
```

Comment:
What does this test have to do with dictionaries?

Reply:
I was mainly trying to validate the updated C Data interface on `StructVectorLoader` and `StructVectorUnloader`. Is that wrong?

Reply:
@lidavidm any thoughts about this?